nvflare/private/fed/server/job_cmds.py

```diff
@@ -407,12 +407,20 @@ def abort_job(self, conn: Connection, args: List[str]):
         try:
             job_id = conn.get_prop(self.JOB_ID)
             with engine.new_context() as fl_ctx:
-                message = job_runner.stop_run(job_id, fl_ctx)
-                if message:
-                    conn.append_error(message, meta=make_meta(MetaStatusValue.INTERNAL_ERROR, message))
-                else:
-                    conn.append_string("Abort signal has been sent to the server app.")
+                job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
+                job = job_manager.get_job(job_id, fl_ctx)
+                if job.meta.get(JobMetaKey.STATUS) in [RunStatus.SUBMITTED, RunStatus.DISPATCHED]:
```

Comment:
I wonder whether we need DISPATCHED status. Does it serve us any good? What if the system crashed when the job is left at DISPATCHED status? If the job is still in SUBMITTED status, it will get a chance to be rescheduled.

Reply:
DISPATCHED status is a transition state after the job deployed to all the parties. If for some reasons the job got deployed but could not get into RUNNIG state, we want to have manual investigation into this situation, instead of leaving it to SUBMITTED, then may recursively get into this again.